lib/utils.js

```diff
@@ -1,52 +1,113 @@
-'use strict';
+'use strict'
 
-var Cache = require('./cache');
-var Mustache = require('mustache');
-var Needle = require('needle');
-var Winston = require('winston');
+const Cache = require('./cache')
+const Mustache = require('mustache')
+const Needle = require('needle')
+const Winston = require('winston')
+const CronJob = require('cron').CronJob
+const Routes = require('./routes')
+const URL = require('url')
+const removeRouteDep = require('express-remove-route')
 
-var template = '{"title":"{{title}}","headers":{{{headers}}},"rows":[{{{row}}}],"width":{{width}},"height":1}';
-Mustache.parse(template);
+const template = '{"title":"{{title}}","headers":{{{headers}}},"rows":[{{{row}}}],"width":{{width}},"height":1}'
+Mustache.parse(template)
 
 // Fetch all data and add it to the cache
-exports.populateCache = function populateCache(url) {
-  Winston.info('Populating cache');
+function populateCache (url) {
+  Winston.info('Populating cache')
 
   // Fetch the facility data
-  Needle.get(url, function(err, res) {
+  Needle.get(url, function (err, res) {
     if (err) {
-      return Winston.error('Fetching facility data failed', {err: err.message});
+      return Winston.error('Fetching facility data failed', {err: err.message})
     }
 
     if (res.statusCode !== 200) {
-      return Winston.error('Non-200 status code: %d', res.statusCode);
+      return Winston.error('Non-200 status code: %d', res.statusCode)
     }
 
-    var title = res.body.title;
-    var headers = res.body.headers;
+    const title = res.body.title
+    const headers = res.body.headers
 
-    var batch = Cache.batch();
+    const batch = Cache.batch()
 
     // Add each facility to the cache
-    res.body.rows.forEach(function(row) {
-      var key = row[0];
+    res.body.rows.forEach(function (row) {
+      const key = row[0]
       if (!key) {
-        return Winston.error('Facility missing value', {facility: row});
+        return Winston.error('Facility missing value', {facility: row})
       }
-      var body = Mustache.render(template, {
+      const body = Mustache.render(template, {
         title: title,
         headers: JSON.stringify(headers),
         row: JSON.stringify(row),
         width: headers.length
-      });
-      batch.put(key, body);
-    });
+      })
+      batch.put(key, body)
```

Comment:
The cache isn't big at all at the moment, so i think this is fine for now, This is only for one level of an organisation unit, across +- 10 countries. I'm just wondering if this might become an issue at a later stage if bigger instances of DHIS are introduced, Something like DATIM that stretches over global facilities.

I guess no need for pre-optimisation before its needed :)